wsd/LOOLWSD.cpp

```diff
@@ -1708,7 +1708,7 @@ bool LOOLWSD::checkAndRestoreForKit()
                 }
                 else
                 {
-                    LOG_ERR("Forkit process [" << pid << "] " <<
+                    LOG_FTL("Forkit process [" << pid << "] " <<
```

Comment:
It creates a denied service, and a core dump is created. It is a very important process (it is rare the "forkit" process to die), anyway for debugging purposes if we grep LOiG_FTL and check how this rare process die and if it recovered will give point us to investigate the causes. It should be LOG_FTL

Reply:
I would expect wsd to -almost-never- have a FTL failure; if WSD fails - the service is down, which is a catastrophy for availability. I'm pleased that Ash is going over these with a fine tooth comb. Fair enough for LOG_FTL in the Kit process - that's just one document - but to loose every current document, and all document data being edited because the forkit crashed is not proportionate or reasonable IMHO.

Reply:
> It creates a denied service [...]

I don't think I understand this. Can you explain what you have in mind please?

If forkit dies (say, because of memory being low, or we run out file-handles and crash, or something like that), we will create a new instance. There is no "denied service." The process of recovery is seamless and there is no interruption of service at all. If we can't spawn forkit and/or child kits, then we log errors that we can't service document load requests. But I don't see any impact on the service because forkit crashed. We have tests for this scenario and we recover without any issue from such forkit crashes.

Reply:
> I don't think I understand this. Can you explain what you have in mind please?
> 
> If forkit dies (say, because of memory being low, or we run out file-handles and crash, or something like that), we will create a new instance. There is no "denied service." The process of recovery is seamless and there is no interruption of service at all. If we can't spawn forkit and/or child kits, then we log errors that we can't service document load requests. But I don't see any impact on the service because forkit crashed. We have tests for this scenario and we recover without any issue from such forkit crashes.

Well, if the "forkit" process crash until it recovers, there is a small delta time "denied service", depending of the pre-spawn parameters, best scenario. The worst case, no recovery the "forkit" and all children are ripped.